src/proxy/http3/Http3Frame.cc

```diff
@@ -280,10 +280,20 @@ Http3SettingsFrame::Http3SettingsFrame(const uint8_t *buf, size_t buf_len, uint3
     size_t   id_len  = QUICVariableInt::size(buf + len);
     uint16_t id      = QUICIntUtil::read_QUICVariableInt(buf + len, buf_len - len);
     len             += id_len;
+    if (len > buf_len) {
```

Comment:
It should probably be `len >= buf_len`. Having only id is malformed, and current code (even with your fix) can cause buffer-over-run on `size_t value_len = QUICVariableInt::size(buf + len);`.

Reply:
Yes. I think even the reading is dangerous. I should give up before the reading